### PR TITLE
placement: fix alter placement policy cannot update the relative ranges policy

### LINKED ISSUE #51712: Alter placement policy cannot update the relative ranges policy.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
CREATE PLACEMENT POLICY `evict_sata_dw` CONSTRAINTS="[-disk=sata-new, -disk=dw-ssd]"
ALTER RANGE global PLACEMENT POLICY `evict_sata_dw`.
ALTER PLACEMENT POLICY evict_sata_dw CONSTRAINTS="[-disk=sata-new, -disk=dw-ssd]" SURVIVAL_PREFERENCES="[zone, dc, host]"
```

### 2. What did you expect to see? (Required)

global rule will update the location labels.
### 3. What did you see instead (Required)
no update

```
root@prod-dw-tipd6-f001:~# tiup ctl:v7.5.1 pd config placement-rules rule-bundle get pd
Starting component ctl: /root/.tiup/components/ctl/v7.5.1/ctl pd config placement-rules rule-bundle get pd
{
  "group_id": "pd",
  "group_index": 0,
  "group_override": false,
  "rules": [
    {
      "group_id": "pd",
      "id": "default",
      "start_key": "",
      "end_key": "",
      "role": "voter",
      "is_witness": false,
      "count": 3,
      "location_labels": [
        "zone",
        "dc",
        "host"
      ],
      "version": 1
    }
  ]
}

root@prod-dw-tipd6-f001:~# tiup ctl:v7.5.1 pd config placement-rules rule-bundle get TiDB_GLOBAL
Starting component ctl: /root/.tiup/components/ctl/v7.5.1/ctl pd config placement-rules rule-bundle get TiDB_GLOBAL
{
  "group_id": "TiDB_GLOBAL",
  "group_index": 20,
  "group_override": true,
  "rules": [
    {
      "group_id": "TiDB_GLOBAL",
      "id": "evict_sata_dw_rule_0",
      "start_key": "",
      "end_key": "",
      "role": "voter",
      "is_witness": false,
      "count": 3,
      "label_constraints": [
        {
          "key": "disk",
          "op": "notIn",
          "values": [
            "sata-new"
          ]
        },
        {
          "key": "disk",
          "op": "notIn",
          "values": [
            "dw-ssd"
          ]
        },
        {
          "key": "engine",
          "op": "notIn",
          "values": [
            "tiflash"
          ]
        }
      ],
      "create_timestamp": 1710164464
    }
```

need `ALTER RANGE global PLACEMENT POLICY evict_sata_dw.` again then update.

### 4. What is your TiDB version? (Required)
7.5.1

### LINKED ISSUE #52257: Can't drop placement policy if policyName is part of the policyName used by range
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
create placement policy mypolicy followers=4;
create placement policy mypolicy2 followers=2;
alter range global placement policy mypolicy2;
drop placement policy mypolicy;
```

### 2. What did you expect to see? (Required)

drop mypolicy return success.

### 3. What did you see instead (Required)

```
mysql> create placement policy mypolicy followers=4;
Query OK, 0 rows affected (0.11 sec)

mysql> create placement policy mypolicy2 followers=2;
Query OK, 0 rows affected (0.11 sec)

mysql> alter range global placement policy mypolicy2;
Query OK, 0 rows affected (0.00 sec)

mysql> drop placement policy mypolicy;
ERROR 8241 (HY000): Placement policy 'mypolicy' is still in use
```

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by jiyfhust**
### What problem does this PR solve?

Issue Number: close #51712 close #52257

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
@jiyfhust Please sync with master.

**Review by nolouch — APPROVED** [review]
lgtm. thanks!

**Review by xhebox — APPROVED** [review]

**Comment by jiyfhust** [comment]
> please assign [benjamin2037](https://github.com/benjamin2037) for approval.

PTAL @zimulala @Benjamin2037

File: pkg/executor/show_placement.go [hunk 1]
```diff
@@ -268,39 +268,26 @@ func (e *ShowExec) fetchAllPlacementPolicies() error {
 }
 
 func (e *ShowExec) fetchRangesPlacementPlocy(ctx context.Context) error {
```
Comment (zimulala) [inline pkg/executor/show_placement.go:270]: Could we add a test for this change?

Comment (jiyfhust) [reply pkg/executor/show_placement.go:270]: PTAL @zimulala https://github.com/pingcap/tidb/pull/52254/commits/378227dd96c6eda79a769b84aef2cd329ed14116

**Review by zimulala — APPROVED** [review]

**Pull request merged**